gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/task/HiveConverterUtils.java

```diff
@@ -0,0 +1,437 @@
+   * @param inputTblName source table name
+   * @param tblName New Hive table name
+   * @param tblLocation New hive table location
+   * @param optionalDbName Optional DB name, if not specified it defaults to 'default'
+   */
+  public static String generateCreateDuplicateTableDDL(
+      String inputDbName,
+      String inputTblName,
+      String tblName,
+      String tblLocation,
+      Optional<String> optionalDbName) {
+
+    Preconditions.checkArgument(StringUtils.isNotBlank(tblName));
+    Preconditions.checkArgument(StringUtils.isNotBlank(tblLocation));
+
+    String dbName = optionalDbName.isPresent() ? optionalDbName.get() : "default";
+
+    return String.format("CREATE EXTERNAL TABLE IF NOT EXISTS `%s`.`%s` LIKE `%s`.`%s` LOCATION %n  '%s' %n",
+        dbName, tblName, inputDbName, inputTblName, tblLocation);
+  }
+
+  public static String generateTableCopy(
+      String inputTblName,
+      String outputTblName,
+      Optional<String> optionalInputDbName,
+      Optional<String> optionalOutputDbName,
+      Optional<Map<String, String>> optionalPartitionDMLInfo,
+      Optional<Boolean> optionalOverwriteTable,
+      Optional<Boolean> optionalCreateIfNotExists) {
+    Preconditions.checkArgument(StringUtils.isNotBlank(inputTblName));
+    Preconditions.checkArgument(StringUtils.isNotBlank(outputTblName));
+
+    String inputDbName = optionalInputDbName.isPresent() ? optionalInputDbName.get() : DEFAULT_DB_NAME;
+    String outputDbName = optionalOutputDbName.isPresent() ? optionalOutputDbName.get() : DEFAULT_DB_NAME;
+    boolean shouldOverwriteTable = optionalOverwriteTable.isPresent() ? optionalOverwriteTable.get() : true;
+    boolean shouldCreateIfNotExists = optionalCreateIfNotExists.isPresent() ? optionalCreateIfNotExists.get() : false;
+
+    // Start building Hive DML
+    // Refer to Hive DDL manual for explanation of clauses:
+    // https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DML#LanguageManualDML-InsertingdataintoHiveTablesfromqueries
+    StringBuilder dmlQuery = new StringBuilder();
+
+    // Insert query
+    if (shouldOverwriteTable) {
+      dmlQuery.append(String.format("INSERT OVERWRITE TABLE `%s`.`%s` %n", outputDbName, outputTblName));
+    } else {
+      dmlQuery.append(String.format("INSERT INTO TABLE `%s`.`%s` %n", outputDbName, outputTblName));
+    }
+
+    // Partition details
+    dmlQuery.append(partitionKeyValues(optionalPartitionDMLInfo));
+
+    // If not exists
+    if (shouldCreateIfNotExists) {
+      dmlQuery.append(" IF NOT EXISTS \n");
+    }
+    dmlQuery.append(String.format("SELECT * FROM `%s`.`%s`", inputDbName, inputTblName));
+    if (optionalPartitionDMLInfo.isPresent()) {
+      if (optionalPartitionDMLInfo.get().size() > 0) {
+        dmlQuery.append(" WHERE ");
+        boolean isFirstPartitionSpec = true;
+        for (Map.Entry<String, String> partition : optionalPartitionDMLInfo.get().entrySet()) {
+          if (isFirstPartitionSpec) {
```

Comment:
It is more understandable if you use a `Joiner` instead of keeping state of whether it is the first token or not.